packages/webchat/src/core/socket.tsx

```diff
@@ -1,73 +1,95 @@
-import { Config, StudioConnector } from '../typings'
+import { Message, MessagingSocket, UserCredentials } from '@botpress/messaging-socket'
+import { Config } from '../typings'
 
 export default class BpSocket {
-  private events: any
-  private userIdScope?: string
+  public socket: MessagingSocket
   private chatId: string | undefined
+  private waitingForUser?: Promise<void>
 
   public onClear!: (event: any) => void
   public onMessage!: (event: any) => void
   public onTyping!: (event: any) => void
   public onData!: (event: any) => void
   public onUserIdChanged!: (userId: string) => void
 
-  constructor(bp: StudioConnector, config: Config) {
-    this.events = bp?.events
-    this.userIdScope = config.userIdScope
+  constructor(config: Config) {
     this.chatId = config.chatId
+    this.socket = new MessagingSocket({ url: config.url, clientId: config.clientId })
   }
 
   public setup() {
-    if (!this.events) {
-      return
-    }
-
-    // Connect the Botpress Web Socket to the server
-    this.events.setup(this.userIdScope)
+    this.socket.on('message', this.onMessage)
 
+    /*
     this.events.on('guest.webchat.clear', this.onClear)
     this.events.on('guest.webchat.message', this.onMessage)
     this.events.on('guest.webchat.typing', this.onTyping)
     this.events.on('guest.webchat.data', this.onData)
+    */
 
     // firehose events to parent page
-    this.events.onAny(this.postToParent)
+    // TODO: why do we need this
+    // this.events.onAny(this.postToParent)
```

Comment:
This way you can add a custom handler in your page to handle any incoming events with `window.addEventListener('message')`